Codecademy/gamut
# fix(Icons): remove default title

### Overview


Removes the default title from icons (per this accessibility ticket). Also adds a discriminated union type that ensures icons with aria-labels are not hidden


### PR Checklist

- [x] Related to JIRA ticket: [gm-594]

#### Testing Instructions



* check out Icons story - they should not have default titles
* run the branch locally - check out the new type errors when using an Icon with an aria-label without disabling aria-hidden

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Mono Link](https://github.com/codecademy-engineering/Codecademy/pull/37552)  |TBD |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/5889)  | [Portal Env] TBD  |
